cmdpolling/polling.go

```diff
@@ -80,41 +86,69 @@ func cmdStop(c *cli.Context) error {
+func pingRoutine(ctx context.Context, c *cli.Context, longTimePeriod int64, shortTimePeriod int64) {
 	log.Debug("pingRoutine")
 
 	formatter := format.GetFormatter()
 	pollingSvc := cmd.WireUpPolling(c)
 
+	// initialization
 	isRunningCommandRoutine := false
-	t := time.NewTicker(time.Duration(pollingPingTimingInterval) * time.Second)
+	longTicker := time.NewTicker(time.Duration(longTimePeriod) * time.Second)
+	shortTicker := time.NewTicker(time.Duration(shortTimePeriod) * time.Second)
+	currentTicker := longTicker
 	for {
-		log.Debug("Requesting for candidate commands status")
-		ping, status, err := pollingSvc.Ping()
-		if err != nil {
-			formatter.PrintError("Couldn't receive polling ping data", err)
-		} else {
-			// One command is available, and no process running
-			if status == 201 && ping.PendingCommands && !isRunningCommandRoutine {
-				log.Debug("Detected a candidate command")
-				isRunningCommandRoutine = true
-				go processingCommandRoutine(pollingSvc, formatter)
+		// no need to request until current command ends
+		if !isRunningCommandRoutine {
+			log.Debug("Requesting for candidate commands status")
+			ping, status, err := pollingSvc.Ping()
+			if err != nil {
+				formatter.PrintError("Couldn't receive polling ping data", err)
+				// in low level error, should the ticker set as the the longest time interval?
+			} else {
+				// One command is available
+				if status == 201 {
+					if ping.PendingCommands {
+						log.Debug("Detected a candidate command")
+						isRunningCommandRoutine = true
+						// set short interval timing!?
+						// - by default this implies next interval timing sil be short.
+						// - If command routine has a over delay, long interval timing will be assigned later
+						if currentTicker != shortTicker {
+							log.Debug("Ticker assigned: Short")
+							shortTicker = time.NewTicker(time.Duration(shortTimePeriod) * time.Second)
+							currentTicker = shortTicker
+						}
+						go processingCommandRoutine(pollingSvc, formatter)
+					} else {
+						// Since no pending command, long interval timing
+						log.Debug("Ticker assigned: Long")
+						currentTicker = time.NewTicker(time.Duration(longTimePeriod) * time.Second)
+					}
+				}
 			}
 		}
 
-		select {
-		case <-commandProcessed:
-			isRunningCommandRoutine = false
-		default:
-		}
+		log.Debug("Waiting...", currentTicker)
 
 		select {
-		case <-t.C:
+		case <-currentTicker.C:
 		case <-ctx.Done():
 			log.Debug(ctx.Err())
 			log.Debug("closing polling")
 			return
 		}
+
+		select {
+		case <-commandProcessed:
+			isRunningCommandRoutine = false
+		default:
+			// if command routine is currently running and short interval timing runs out, a long interval timing is assigned
+			if isRunningCommandRoutine && currentTicker == shortTicker {
+				log.Debug("Ticker re-assigned: Long")
+				currentTicker = time.NewTicker(time.Duration(longTimePeriod) * time.Second)
```

Comment:
This select would disappear (moved and simplified to the first one of the loop), but at this point if the currentTicker is not the longTicker we still need to do the following before the end of the for loop:
- Stop the current (short) ticker to release associated resources ( see https://godoc.org/time#NewTicker)
- Set the current ticker to the long ticker